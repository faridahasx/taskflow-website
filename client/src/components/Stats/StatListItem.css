.stat-li {
    height: 40px;
    padding: 0 20px;
    white-space: nowrap;
    align-items: center;
}

.stat-key {
    width: 50%;
    min-width: 200px;
    font-weight: 500;
}

.stat-value {
    width: 50%;
}


.stat-li:first-child,
.stat-li:first-child .stat-key,
.stat-li:last-child,
.stat-li:last-child .stat-key {
    font-weight: 600;
}

@media all and (min-width: 400px) {
    .stat-li {
        border-bottom: 1px solid rgba(128, 128, 128, 0.363);
    }

    .stat-li:last-child {
        border-top: 1px solid rgb(128, 128, 128);
    }

    .stat-li:first-child {
        border-bottom-color: rgb(128, 128, 128);
    }
}