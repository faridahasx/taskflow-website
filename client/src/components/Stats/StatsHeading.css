#stats-heading {
    width: 100%;
    padding-top: 8px;
    align-items: center;
    justify-content: space-between;
}

#stats-dt-container {
    font-weight: 500;
    font-size: 15px;
    font-style: italic;
}


.datetime {
    margin: 0 5px;
}

@media all and (max-width: 380px) {
    #stats-heading {
        align-items: start;
        flex-direction: column;
    }

    .datetime {
        margin: 5px 5px 5px 0;
    }

    .datetime:nth-child(2) {
        margin-right: 5px;
    }
}