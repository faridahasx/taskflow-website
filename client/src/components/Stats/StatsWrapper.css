#stats {
    width: 100%;
    min-height: calc(var(--vh) - 170px);
    margin-top: 40px;
    padding: 40px 0;
    justify-content: center;
    align-items: center;
}

#stats .icon {
    border-color: rgba(0, 0, 0, 0);
}

#stats-wrapper {
    width: calc(100vw - 40px);
    max-width: 800px;
    min-width: calc(var(--min-width) - 20px);
    border-radius: 10px;
    background-color: rgba(255, 255, 255, 0.922);
    box-shadow: 0 4px 8px -10px rgba(100, 94, 94, 0.604), 0 6px 20px 0 rgba(0, 0, 0, 0.275);
}



.stats-empty {
    height: 280px;
}


#stats-top {
    align-items: start;
    padding: 5px 10px;
    width: 100%;
}


#stats-top h1 {
    color: var(--main-color-dark);
    font-size: 19px;
}


#stats-listing {
    width: 100%;
    overflow: auto;
    border-top: 1px solid rgb(128, 128, 128);
}

#stats-listing::-webkit-scrollbar-track {
    margin: 0 10px;
}



@media all and (min-width: 900px) {
    #stats-wrapper {
        margin-left: 50px;
    }
}