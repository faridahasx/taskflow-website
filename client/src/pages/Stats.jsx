/**
 * Stats Component
 *
 * This component represents the statistics page,
 * rendering StatsContainer inside Layout
 * to display relevant statistical information about tasks.
 *
 */

// Imports
import StatsContainer from "../containers/StatsContainer";
import Layout from "../components/Layout/Layout";

const Stats = () => {
  return (
    <Layout>
      <StatsContainer />
    </Layout>
  );
};

export default Stats;
