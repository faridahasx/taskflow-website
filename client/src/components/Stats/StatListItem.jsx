// External imports
import PropTypes from "prop-types";

// Styles
import "./StatListItem.css";

const StatListItem = ({ statKey, statValue }) => {
  return (
    <li className="stat-li flex">
      <span className="stat-key flex">{statKey}:</span>
      <span className="stat-value flex">{statValue}</span>
    </li>
  );
};

StatListItem.propTypes = {
  statKey: PropTypes.string.isRequired,
  statValue: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
    .isRequired,
};
export default StatListItem;
