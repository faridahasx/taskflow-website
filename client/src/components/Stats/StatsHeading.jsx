import "./StatsHeading.css";

const StatsHeading = ({ start, end }) => {
  return (
    <h1 id="stats-heading" className="flex">
      Tasks:
      {start && (
        <span id="stats-dt-container" className="center">
          <span className="center datetime">{start}</span> -
          <span className="center datetime">{end}</span>
        </span>
      )}
    </h1>
  );
};

export default StatsHeading;
