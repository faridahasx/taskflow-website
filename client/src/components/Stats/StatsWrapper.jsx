// External imports
import { useMemo } from "react";
import PropTypes from "prop-types";
// Hooks
import useNetworkStatus from "../../hooks/useNetworkStatus";
// Components
import OfflineContent from "../../components/OfflineContent";
import CircularLoading from "../../components/Loading/CircularLoading";
import FiltersNav from "../../components/NavigationItem/FiltersNav";
import StatListItem from "./StatListItem";
import StatsHeading from "./StatsHeading";
// Styles
import "./StatsWrapper.css";

const StatsWrapper = ({ stats, loading, isLogged, start, end }) => {
  const statKeys = useMemo(() => stats && Object.keys(stats), [stats]);
  const isOnline = useNetworkStatus();

  return (
    <div id="stats" className="column">
      {!isOnline && stats === null ? (
        <OfflineContent />
      ) : (
        <div id="stats-wrapper" className="column">
          <>
            <div id="stats-top" className="flex">
              <FiltersNav />
              <StatsHeading start={start} end={end} />
            </div>

            {isLogged === false ? (
              <span className="center stats-empty">
                Sign in to unlock this feature
              </span>
            ) : loading || stats === null ? (
              <span className="center stats-empty">
                <CircularLoading />
              </span>
            ) : (
              stats !== null &&
              (statKeys.length ? (
                <ul id="stats-listing" className="">
                  {statKeys.map((s) => (
                    <StatListItem key={s} statKey={s} statValue={stats[s]} />
                  ))}
                </ul>
              ) : (
                <span className="center stats-empty">No available data</span>
              ))
            )}
          </>
        </div>
      )}
    </div>
  );
};

StatsWrapper.propTypes = {
  stats: PropTypes.any,
  start: PropTypes.string.isRequired,
  end: PropTypes.string.isRequired,
  loading: PropTypes.bool.isRequired,
};

export default StatsWrapper;
