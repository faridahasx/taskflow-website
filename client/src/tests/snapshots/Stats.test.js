import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import Stats from "../../pages/Stats";
import mockInitialState, { mockStore } from "../mocks/mockReduxState";

test("Stats snapshot", () => {
  const { asFragment } = render(
    <Provider store={mockStore(mockInitialState)}>
      <BrowserRouter>
        <Stats />
      </BrowserRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot("Stats");
});
