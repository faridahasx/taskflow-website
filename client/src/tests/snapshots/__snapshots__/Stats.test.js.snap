// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats snapshot: Stats 1`] = `
<DocumentFragment>
  <header
    class="flex"
    id="header"
  >
    <span
      class="flex"
      id="header-first"
    >
      <button
        class="icon "
        title="Open categories"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </button>
      <a
        class="center"
        data-testid="header-home-link"
        href="/"
      >
        <div
          class="logo center"
          title="TaskFlow"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MovingIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19.71 9.71 22 12V6h-6l2.29 2.29-4.17 4.17c-.39.39-1.02.39-1.41 0l-1.17-1.17c-1.17-1.17-3.07-1.17-4.24 0L2 16.59 3.41 18l5.29-5.29c.39-.39 1.02-.39 1.41 0l1.17 1.17c1.17 1.17 3.07 1.17 4.24 0l4.19-4.17z"
            />
          </svg>
          <span
            class="logo-task"
          >
            Task
          </span>
          <span
            class="logo-flow"
          >
            Flow
          </span>
        </div>
      </a>
    </span>
    <span
      class="flex"
      id="header-second"
    >
      <a
        class="icon"
        data-testid="header-stats-link"
        href="/stats"
        title="Stats"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="BarChartIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M4 9h4v11H4zm12 4h4v7h-4zm-6-9h4v16h-4z"
          />
        </svg>
      </a>
      <button
        class="icon "
        title="Sign In"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="LoginOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 7 9.6 8.4l2.6 2.6H2v2h10.2l-2.6 2.6L11 17l5-5-5-5zm9 12h-8v2h8c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-8v2h8v14z"
          />
        </svg>
      </button>
    </span>
  </header>
  <main>
    <div
      class="categories-slider "
      data-testid="categories"
      id="categories-dropdown"
    >
      <div
        class="categories "
      >
        <div
          class="center"
          id="categories-list-container"
        >
          <ol
            id="categories-list"
          >
            <li
              class="category-li flex"
            >
              <a
                class="flex"
                href="/"
              >
                <span
                  class="text-overflow"
                >
                  All
                </span>
                <span>
                  (5)
                </span>
              </a>
            </li>
            <li
              class="category-li flex"
            >
              <a
                class="flex text-overflow"
                href="/?categories=Tasks"
              >
                <span
                  class="text-overflow"
                  title="Tasks"
                >
                  Tasks
                </span>
                <span
                  title="5 tasks"
                >
                  (5)
                </span>
              </a>
              <button
                class="icon "
                data-testid="Tasks-menu"
                title="More"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="MoreVertOutlinedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </button>
            </li>
          </ol>
        </div>
        <button
          class="center"
          id="add-category"
          title="Add new category"
        >
          <span
            class="center"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="AddIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
              />
            </svg>
          </span>
          New Category
        </button>
      </div>
    </div>
    <div
      class="column"
      id="stats"
    >
      <div
        class="column"
        id="stats-wrapper"
      >
        <div
          class="flex"
          id="stats-top"
        >
          <span
            class="nav-item center"
          >
            <button
              class="icon "
              title="Filters"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="FilterListIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
                />
              </svg>
            </button>
          </span>
          <h1
            class="flex"
            id="stats-heading"
          >
            Tasks:
          </h1>
        </div>
        <span
          class="center stats-empty"
        >
          Sign in to unlock this feature
        </span>
      </div>
    </div>
  </main>
  <footer
    class="center"
  >
    <div
      class="center"
      id="footer-div"
    >
      <a
        aria-label="Go to home page"
        href="/"
        id="footer-link"
        title="Home"
      >
        <div
          class="logo center"
          title="TaskFlow"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MovingIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19.71 9.71 22 12V6h-6l2.29 2.29-4.17 4.17c-.39.39-1.02.39-1.41 0l-1.17-1.17c-1.17-1.17-3.07-1.17-4.24 0L2 16.59 3.41 18l5.29-5.29c.39-.39 1.02-.39 1.41 0l1.17 1.17c1.17 1.17 3.07 1.17 4.24 0l4.19-4.17z"
            />
          </svg>
          <span
            class="logo-task"
          >
            Task
          </span>
          <span
            class="logo-flow"
          >
            Flow
          </span>
        </div>
      </a>
      <a
        href="https://farida-mu.vercel.app/#contact"
        id="dev-info"
        target="_blank"
        title="Farida Hasanova"
      >
        Contact the developer
      </a>
    </div>
  </footer>
</DocumentFragment>
`;
